From af0517301677b4206c605caaef25f5d57a31b5b8 Mon Sep 17 00:00:00 2001
From: "A. Wilcox" <AWilcox@Wilcox-Tech.com>
Date: Fri, 9 Jun 2017 00:26:16 -0500
Subject: [PATCH] catopen: set errno to EOPNOTSUPP

Per 1003.1-2008 (2016 ed.), catopen must set errno on failure.

We set errno to EOPNOTSUPP because musl does not currently support
message catalogues.
---
 src/locale/catopen.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/locale/catopen.c b/src/locale/catopen.c
index 4423c4d9..3fbc7717 100644
--- a/src/locale/catopen.c
+++ b/src/locale/catopen.c
@@ -1,6 +1,8 @@
 #include <nl_types.h>
+#include <errno.h>
 
 nl_catd catopen (const char *name, int oflag)
 {
+	errno = EOPNOTSUPP;
 	return (nl_catd)-1;
 }
-- 
2.13.0

